corehq/apps/domain/models.py

```diff
@@ -822,6 +823,11 @@ def save_snapshot(self, share_reminders, copy_by_id=None):
             except NameUnavailableException:
                 return None
             copy.is_snapshot = True
+            for snapshot in self.snapshots():
```

Comment:
Note that if you do this approach, you should use ES only to get the ids, then use `iter_docs` to get the couch docs to save - you don't want to be reading from ES and saving to couch.